### PR TITLE
Change company interest page

### PR DESCRIPTION
**Pull request by Lundis1**
# Description

Changed text on some og the descriptions on the company interest form, and made some of the fields mandatory, as requested by Bedkom. 

# Testing

- [x ] I have thoroughly tested my changes.

---

Resolves ABA-306

### CODE REVIEW

File: app/routes/companyInterest/components/CompanyInterestPage.tsx [hunk 1]
```diff
@@ -291,18 +142,36 @@ const EventBox = ({
...
-        label={EVENT_TYPES[eventToString(key)][language]}
-        type="checkbox"
-        component={CheckBox.Field}
-        normalize={(v) => !!v}
-      />
-    ))}
-  </Flex>
+  <FormSpy subscription={{ values: true }}>
+    {(props) => {
+      const newFields = fields.map((field) => field);
```
Comment (eikhr) [inline app/routes/companyInterest/components/CompanyInterestPage.tsx:147]: Perhaps leave a comment explaining this. It looks a bit unnecessary if you're not too familiar with `react-final-form`.
I also think maybe "filteredFields" would be a more descriptive name for this variable, as nothing new is being added to the fields.

File: app/routes/companyInterest/components/CompanyInterestPage.tsx [hunk 2]
```diff
@@ -291,18 +142,36 @@ const EventBox = ({
...
+      if (props.values.officeInTrondheim !== 'yes') {
+        let indexOfCompanyToCompany = -1;
+        fields.forEach((field, index) => {
+          if (fields.value[index].name == 'company_to_company') {
+            indexOfCompanyToCompany = index;
+          }
+        });
+        if (indexOfCompanyToCompany != -1) {
+          newFields.splice(indexOfCompanyToCompany, 1);
```
Comment (eikhr) [inline app/routes/companyInterest/components/CompanyInterestPage.tsx:156]: Is the `indexOfCompanyToCompany` variable necessary? Couldn't we just splice the list inside the forEach-loop?

File: app/routes/companyInterest/components/CompanyInterestPage.tsx [hunk 3]
```diff
@@ -449,12 +318,22 @@ type Props = {
...
   contactPerson: [required()],
   mail: [required(), isEmail()],
   phone: [required()],
-  comment: [required()],
+  aboutTheCompany: [required()],
+  companyType: [required()],
+  officeInTrondheim: [required()],
+  events: [required()],
+  breakfastTalkComment: [requiredIfEventType('breakfast_talk')],
 });
```
Comment (eikhr) [inline app/routes/companyInterest/components/CompanyInterestPage.tsx:337]: Seems like you might have forgotten to validate the comments of the other event types

File: app/routes/companyInterest/components/CompanyInterestPage.tsx [hunk 4]
```diff
@@ -699,15 +481,24 @@ const CompanyInterestPage = (props: Props) => {
...
-                </RadioButtonGroup>
+                </MultiSelectGroup>
+
+                <FormSpy>
+                  {(props) => {
+                    return (
+                      <pre>{JSON.stringify(props.values, undefined, 4)}</pre>
+                    );
+                  }}
+                </FormSpy>
```
Comment (eikhr) [inline app/routes/companyInterest/components/CompanyInterestPage.tsx:495]: 👀

**Review by eikhr — CHANGES_REQUESTED** [review]
Love all the cleanup you've done in addition to the necessary changes!
Some cleanup is needed before merge:)

File: app/components/Form/MultiSelectGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                <RenderErrorMessage
+                  key={submitError}
+                  error={submitError}
+                  fieldName={name}
+                />
+              </div>
+            );
+          }
+
+          return <div></div>;
```
Comment (ivarnakken) [inline app/components/Form/MultiSelectGroup.tsx:51]: Any reason for returning an empty div? It will just add to the DOM, which is unnecessary

File: app/routes/companyInterest/components/CompanyIntrestPageTranslations.tsx
Comment (ivarnakken) [inline app/routes/companyInterest/components/CompanyIntrestPageTranslations.tsx:1]: Very nitpick, **and only for later**, since the file does not contain any JSX it's not really a `.tsx` file, but just a`.ts` file

Comment (ivarnakken) [reply app/routes/companyInterest/components/CompanyIntrestPageTranslations.tsx:1]: Also the name is a bit long, since we're already inside a `companyInterest` directory, naming it just `translations.ts` would be sufficient

**Review by eikhr — APPROVED** [review]

File: app/routes/companyInterest/components/CompanyIntrestPageTranslations.tsx [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+      */
+  /*   collaboration_revue: {
+        norwegian: 'Samarbeid med Revyen**',
+        english: 'Collaboration with the revue**',
+      }, */
+};
+export const TARGET_GRADES = {
+  '1': {
+    norwegian: '1. klasse',
+    english: '1st Years',
```
Comment (ivarnakken) [inline app/routes/companyInterest/components/CompanyIntrestPageTranslations.tsx:147]: Wow it bothers me that `Years` is capitalized and not `klasse`, but I guess that's too nitpicky

File: cypress/e2e/company_interest_spec.js [hunk 7]
```diff
@@ -23,8 +23,6 @@ const createCompanyInterest = () => {
   field('events[0].checked').check();
   field('otherOffers[0].checked').check();
 
-  field('comment').type('random comment');
```
Comment (ivarnakken) [inline cypress/e2e/company_interest_spec.js:26]: Why is this removed? Wasn't this just renamed to `aboutTheCompany`?

**Review by ivarnakken — APPROVED** [review]
🏅

File: app/components/Form/MultiSelectGroup.tsx (hunk 5 above)
Comment (Lundis1) [reply app/components/Form/MultiSelectGroup.tsx:51]: Because the formspy needs to return a component an empty div is the fallback.

File: app/routes/companyInterest/components/CompanyIntrestPageTranslations.tsx
Comment (Lundis1) [reply app/routes/companyInterest/components/CompanyIntrestPageTranslations.tsx:1]: fixed

File: cypress/e2e/company_interest_spec.js (hunk 7 above)
Comment (Lundis1) [reply cypress/e2e/company_interest_spec.js]: though we changed it, and regreted it so changed it back ;)

File: app/routes/companyInterest/components/CompanyIntrestPageTranslations.tsx (hunk 6 above)
Comment (Lundis1) [reply app/routes/companyInterest/components/CompanyIntrestPageTranslations.tsx]: fixed
